### PR TITLE
fix: Check for default Firestore database

### PR DESCRIPTION
**Pull request by NimJay**
### Problem

* Please see Google-internal Buganizer: [b/332863053](http://b/332863053)
* Here's the error:
```
Error creating Database: googleapi: Error 409: Database already exists. Please use another database_id
```
* This Terraform already has mechanism in place to avoid Firestore database naming conflicts.
* But I believe it failed because of the `== 1` in the [code here](https://github.com/GoogleCloudPlatform/terraform-dynamic-javascript-webapp/blob/v1.3.0/infra/main.tf#L18):
```
  firestore_enabled = length(data.google_cloud_asset_resources_search_all.firestore_database.results) == 1 ? true : false
```
* The `== 1` doesn't account for projects with more than one Firestore database.

### Solution 
* We should account for projects with more than one Firestore database.
* We can use the ["query" attribute](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/cloud_asset_resources_search_all#query) to only check for Firestore databases named `(default)`.

### Testing
* In one of the commits inside this pull-request, I temporarily introduced a "pre-existing" Firestore database titled `(default)`. See https://github.com/GoogleCloudPlatform/terraform-dynamic-javascript-webapp/pull/124/commits/01527960a515c872f395b13312cedb8a49953792 (now reverted in this pull-request).
   * [This other commit](https://github.com/GoogleCloudPlatform/terraform-dynamic-javascript-webapp/pull/124/commits/a2f8e3f2476532819df2e58ae63633439ec286c7) (which included the temporarily added pre-existing Firestore database) passed `terraform apply`.

### Additional info
* I assume we're using the name `(default)` because free quota only applies to the `(default)` database. That is, we cannot just give our Firestore database a unique name. [Source](https://firebase.google.com/docs/firestore/manage-databases#console):
```
You can use the [Free quota](https://firebase.google.com/docs/firestore/quotas#free-quota) only with the (default) database.
```

### CODE REVIEW

File: infra/main.tf [hunk 1]
```diff
@@ -14,8 +14,13 @@
...
-  nextauth_url      = "http://${google_compute_global_address.default.address}"
-  firestore_enabled = length(data.google_cloud_asset_resources_search_all.firestore_database.results) == 1 ? true : false
+  nextauth_url                   = "http://${google_compute_global_address.default.address}"
+  has_default_firestore_database = length(data.google_cloud_asset_resources_search_all.default_firestore_database.results) > 0 ? true : false
+}
+
+# Make a GET request to check for Firestore databases named "(default)"
+data "google_cloud_asset_resources_search_all" "default_firestore_database" {
+  query = "name=\"projects/${var.project_id}/databases/(default)\""
 }
```
Comment (NimJay) [inline infra/main.tf:24]: The `google_cloud_asset_resources_search_all` is documented here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/cloud_asset_resources_search_all

It queries the Google Cloud Asset Inventory to find resources within the specified scope (project, folder, or organization).

File: infra/main.tf [hunk 2]
```diff
@@ -318,17 +318,18 @@ data "http" "load_balancer_warm_up" {
 ### Firestore ###
 
 # The following checks Asset Inventory for an existing Firestore database
-data "google_cloud_asset_resources_search_all" "firestore_database" {
+data "google_cloud_asset_resources_search_all" "default_firestore_database" {
   provider = google-beta
   scope    = "projects/${var.project_id}"
+  query    = "displayName:(default)"
```
Comment (glasnt) [inline infra/main.tf:324]: This change makes sense to me, especially with the change of allowing multiple firestore databases. I would want to confirm that this change successfully allows multiple deployments of the same JSS to the one project (and that they are able to operate independently).

Comment (rogerthatdev) [reply infra/main.tf:324]: +1 

I've confirmed that this change successfully allows multiple deployments in 1 project

**Review by rogerthatdev — APPROVED** [review]
Thanks for the detailed PR! LGTM

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**